synapse/storage/databases/main/events_worker.py

```diff
@@ -727,6 +732,20 @@ async def get_missing_events_from_cache_or_db() -> Dict[
                 if not entry.event.rejected_reason
             }
 
+        if get_prev_content:
+            for eventCache in event_entry_map.values():
+                event = eventCache.event
+                if "replaces_state" in event.unsigned:
+                    prev = await self.get_event(
+                        event.unsigned["replaces_state"],
+                        get_prev_content=False,
+                        allow_none=True,
+                    )
+                    if prev:
+                        event.unsigned = dict(event.unsigned)
```

Comment:
We're making a copy of whatever the `event.unsigned` value is here---I guess this is to avoid modifying the events in the cache?

(Presumably it isn't safe to mutate the cached value---I guess we don't want everybodyto see the `prev_content?)

Reply:
TBTH I wasn't quite sure why this is necessary, I was just making sure that I preserved the functionality when updating calling sites-but your speculation sounds plausible to me.